linux_os/guide/system/apparmor/all_apparmor_profiles_in_enforce_complain_mode/ansible/shared.yml

```diff
@@ -0,0 +1,27 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+{{{ ansible_instantiate_variables("var_apparmor_mode") }}}
+
+- name: Ensure all AppArmor Profiles are reloaded
+  ansible.builtin.shell: apparmor_parser -q -r /etc/apparmor.d/
+
+- name: Set all AppArmor profiles to enforce mode 
+  ansible.builtin.shell: aa-enforce /etc/apparmor.d/*
+  when: var_apparmor_mode == 'enforce'
+
+- name: Set all AppArmor profiles to complain mode
+  ansible.builtin.shell: aa-complain /etc/apparmor.d/*
+  when: var_apparmor_mode == 'complain'   
+
+- name: Collect unconfined processes 
+  ansible.builtin.shell: aa-unconfined
+  register: unconfined_processes
+
+- name: Provide details about unconfined processes 
+  ansible.builtin.shell: echo "The process {{ item }} may need to have a profile created or activated for them and then be restarted."
```

Comment:
Here it would be better to use the [assert](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/assert_module.html#ansible-collections-ansible-builtin-assert-module) module.